### PR TITLE
Subsurface input tests

### PR DESCRIPTION
**Pull request by wmww**
Will not compile until the Surface::attach_buffer() PR lands.

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
Let's see if this compiles now...

bors try

**Comment by AlanGriffiths** [comment]
bors try

**Comment by AlanGriffiths** [comment]
bors try

File: tests/subsurfaces.cpp [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+    EXPECT_THAT(client.focused_window(), Ne((wl_surface*)main_surface)) << "input fell through to main surface";
+    EXPECT_THAT(client.focused_window(), Eq((wl_surface*)subsurface));
+    EXPECT_THAT(client.pointer_position(),
+                Eq(std::make_pair(
+                    wl_fixed_from_int(pointer_x - surface_x - subsurface_x),
+                    wl_fixed_from_int(pointer_y - surface_y - subsurface_y))));
+}
+
+// appears to work in normal Mir, but breaks in WLCS
```
Comment (AlanGriffiths) [inline tests/subsurfaces.cpp]: Appears OK in wlcs

Comment (wmww) [reply tests/subsurfaces.cpp]: Ah yes, I fixed that in the wm a while back.

File: tests/subsurfaces.cpp [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    }
+
+    wlcs::Client client{the_server()};
+    wlcs::Surface main_surface{client.create_visible_surface(surface_width, surface_height)};
+    wlcs::Subsurface subsurface{wlcs::Subsurface::create_visible(main_surface, 0, 0, subsurface_width, subsurface_height)};
+};
+
+TEST_F(SubsurfaceTest, subsurface_can_be_created)
+{
+    EXPECT_THAT(&subsurface.parent(), Eq(&main_surface));
```
[the remark this answers is not in the document]
Comment (wmww) [reply tests/subsurfaces.cpp]: No

**Pull request merged**